File: sliding-window/longest-substring-without-repeating-charackters/longest-substring-without-repeating-charackters.cs
public class Solution
{
    public int LengthOfLongestSubstring(string s)
    {
        var last = new Dictionary<char, int>();
        int best = 0, start = 0;

        for (int i = 0; i < s.Length; i++)
        {
            char ch = s[i];
            if (last.TryGetValue(ch, out int prev) && prev >= start)
            {
                start = prev + 1;
            }
            last[ch] = i;
            best = Math.Max(best, i - start + 1);
        }
        return best;
    }
}

File: sliding-window/longest-substring-without-repeating-charackters/longest-substring-without-repeating-charackters.ts
function lengthOfLongestSubstring(s: string): number {
    const last = new Map<string, number>();
    let best = 0, start = 0;

    for (let i = 0; i < s.length; i++) {
        const ch = s[i];
        const prev = last.get(ch);
        if (prev !== undefined && prev >= start) {
            start = prev + 1;
        }
        last.set(ch, i);
        if (i - start + 1 > best) best = i - start + 1;
    }
    return best;
}
